litestar/_openapi/schema_generation/schema.py

```diff
@@ -549,14 +558,98 @@ def create_schema_for_pydantic_model(
         },
         type=OpenAPIType.OBJECT,
         title=_get_type_schema_name(field_type),
-        description=getdoc(field_type) or None,
+    )
+
+
+def create_schema_for_attrs_class(
+    field_type: type[AttrsInstance],
+    generate_examples: bool,
+    plugins: list[OpenAPISchemaPluginProtocol],
+    schemas: dict[str, Schema],
+) -> Schema:
+    """Create a schema object for a given attrs class.
+
+    Args:
+        field_type: An attrs class.
+        generate_examples: Whether to generate examples if none are given.
+        plugins: A list of plugins.
+        schemas: A mapping of namespaces to schemas - this mapping is used in the OA components section.
+
+    Returns:
+        A schema instance.
+    """
+    from attr import NOTHING
+    from attrs import fields_dict
+
+    field_type_hints = get_type_hints(field_type)
+    return Schema(
+        required=sorted(
+            [
+                field_name
+                for field_name, attribute in fields_dict(field_type).items()
+                if attribute.default is NOTHING and not is_optional_union(field_type_hints[field_name])
+            ]
+        ),
+        properties={
+            k: create_schema(
+                field=SignatureField.create(field_type=v, name=k),
+                generate_examples=generate_examples,
+                plugins=plugins,
+                schemas=schemas,
+            )
+            for k, v in field_type_hints.items()
+        },
+        type=OpenAPIType.OBJECT,
+        title=_get_type_schema_name(field_type),
+    )
+
+
+def create_schema_for_struct_class(
+    field_type: type[Struct],
+    generate_examples: bool,
+    plugins: list[OpenAPISchemaPluginProtocol],
+    schemas: dict[str, Schema],
+) -> Schema:
+    """Create a schema object for a given msgspec.Struct class.
+
+    Args:
+        field_type: A msgspec.Struct class.
+        generate_examples: Whether to generate examples if none are given.
+        plugins: A list of plugins.
+        schemas: A mapping of namespaces to schemas - this mapping is used in the OA components section.
+
+    Returns:
+        A schema instance.
+    """
+
+    return Schema(
+        required=sorted(
+            [
+                field.encode_name
+                for field in msgspec_struct_fields(field_type)
+                if field.default is NODEFAULT
+                and field.default_factory is NODEFAULT
+                and not is_optional_union(field.type)
+            ]
+        ),
+        properties={
+            k: create_schema(
+                field=SignatureField.create(field_type=v, name=k),
+                generate_examples=generate_examples,
+                plugins=plugins,
+                schemas=schemas,
+            )
+            for k, v in get_type_hints(field_type).items()
```

Comment:
You'll want to use the `encode_name` here as well. Something like (untested):

```suggestion
            field.encode_name: create_schema(
                field=SignatureField.create(field_type=field.type, name=field.encode_name),
                generate_examples=generate_examples,
                plugins=plugins,
                schemas=schemas,
            )
            for field in msgspec_struct_fields(field_type)
```

For mild efficiency you may want to pull the call to `msgspec_struct_fields` out so it's not called twice (once for `required` and once for `properties`.

A test case that you're doing the right thing would be checking that the output on this struct uses the camelCase names everywhere:

```python
class Test(msgspec.Struct, rename="camel"):
    field_one: int
    field_two: float
```

Reply:
Cheers